homeassistant/components/edl21/sensor.py

```diff
@@ -0,0 +1,177 @@
+        "1-0:16.7.0*255": "Sum active instantaneous power",
+    }
+    _OBIS_BLACKLIST = {
+        # A=129: Manufacturer specific
+        "129-129:199.130.3*255",  # Iskraemeco: Manufacturer
+        "129-129:199.130.5*255",  # Iskraemeco: Public Key
+    }
+
+    def __init__(self, hass, config, async_add_entities) -> None:
+        """Initialize an EDL21 object."""
+        self._registered_obis = set()
+        self._hass = hass
+        self._async_add_entities = async_add_entities
+        self._proto = SmlProtocol(config[CONF_SERIAL_PORT])
+        self._proto.add_listener(self.event, ["SmlGetListResponse"])
+
+    async def connect(self):
+        """Connect to an EDL21 reader."""
+        await self._proto.connect(self._hass.loop)
+
+    def event(self, message_body) -> None:
+        """Handle events from pysml."""
+        assert isinstance(message_body, SmlGetListResponse)
+
+        new_entities = []
+        for telegram in message_body.get("valList", []):
+            obis = telegram.get("objName")
+            if not obis:
+                continue
+
+            if obis in self._registered_obis:
+                async_dispatcher_send(self._hass, SIGNAL_EDL21_TELEGRAM, telegram)
+            else:
+                name = self._OBIS_NAMES.get(obis)
+                if name:
+                    new_entities.append(EDL21Entity(obis, name, telegram))
+                    self._registered_obis.add(obis)
+                elif obis not in self._OBIS_BLACKLIST:
+                    _LOGGER.warning(
+                        "Unhandled sensor %s detected. Please report at "
+                        'https://github.com/home-assistant/home-assistant/issues?q=is%%3Aissue+label%%3A"integration%%3A+edl21"+',
+                        obis,
+                    )
+                    self._OBIS_BLACKLIST.add(obis)
+
+        if new_entities:
+            self._async_add_entities(new_entities, update_before_add=True)
+
+
+class EDL21Entity(Entity):
+    """Entity reading values from EDL21 telegram."""
+
+    def __init__(self, obis, name, telegram):
+        """Initialize an EDL21Entity."""
+        self._obis = obis
+        self._name = name
+        self._telegram = telegram
+        self._min_time = MIN_TIME_BETWEEN_UPDATES
+        self._last_update = utcnow()
+        self._state_attrs = {"status", "valTime", "scaler", "valueSignature"}
+
+    async def async_added_to_hass(self):
+        """Run when entity about to be added to hass."""
+
+        def handle_telegram(telegram):
+            """Update attributes from last received telegram for this object."""
+            if self._obis != telegram.get("objName"):
+                return
+            if self._telegram == telegram:
+                return
+
+            now = utcnow()
+            if now - self._last_update < self._min_time:
+                return
+
+            self._telegram = telegram
+            self._last_update = now
+            self.schedule_update_ha_state()
```

Comment:
Please replace this with `self.async_write_ha_state()`. That will write the state immediately.